Dockerfile

```diff
@@ -35,17 +35,18 @@ RUN apk -U --no-cache add python py-pip \
     && find / -name '*.pyc' -o -name '*.pyo' -exec rm -f {} \;
 
 ADD http://pgoapi.com/pgoencrypt.tar.gz /tmp/pgoencrypt.tar.gz
-RUN apk --no-cache add --virtual .pgoencrypt-dependencies gcc make musl-dev tar \
-    && cat /tmp/pgoencrypt.tar.gz | tar xzf - -C /tmp \
+RUN apk --no-cache add --virtual .pgoencrypt-dependencies gcc make musl-dev \
+    && tar zxf /tmp/pgoencrypt.tar.gz -C /tmp \
     && make -C /tmp/pgoencrypt/src \
     && cp /tmp/pgoencrypt/src/libencrypt.so /usr/src/app/encrypt.so \
     && apk del .pgoencrypt-dependencies \
     && rm -rf /var/cache/apk/* /tmp/pgoencrypt /tmp/pgoencrypt.tar.gz
 
-ADD https://github.com/$BUILD_REPO/archive/$BUILD_BRANCH.tar.gz /tmp
-RUN apk -U --no-cache add --virtual .tar-deps tar \
-    && cat /tmp/$BUILD_BRANCH.tar.gz | tar -zxf - --strip-components=1 -C /usr/src/app \
-    && apk del .tar-deps \
-    && rm /tmp/$BUILD_BRANCH.tar.gz
+ADD https://api.github.com/repos/$BUILD_REPO/git/refs/heads/$BUILD_BRANCH /tmp/pgobot-version
```

Comment:
it order to test new changes... that is why you can also override the BUILD_REPO

In the past there was a COPY . /usr/src/app

the user was able to do it's changes and docker build again

I replaced that with ADD https://github.com/$BUILD_REPO/archive/$BUILD_BRANCH.tar.gz /tmp for various reasons:
- security COPY .  means including auth and unfits... and stuff
- being able to reproduce the build if you know create date  and LABEL build_branch and build_repo and not some local modification that the user might have done...